node/coinstacks/common/api/src/utxo/service.ts

```diff
@@ -182,26 +179,16 @@ export class Service implements Omit<BaseAPI, 'getInfo'>, API {
   }
 
   async handleBlock(hash: string): Promise<Array<BlockbookTx>> {
-    const request: RPCRequest = {
-      jsonrpc: '2.0',
-      id: `getblock-${hash}`,
-      method: 'getblock',
-      params: [hash],
+    try {
+      const { txs = [], totalPages = 1 } = await this.blockbook.getBlock(hash)
+      for (let page = 1; page < totalPages; ++page) {
+        const data = await this.blockbook.getBlock(hash)
```

Comment:
do you need to pass the page number in here?

Reply:
:face_palm: yes